nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/OsCalls.java

```diff
@@ -0,0 +1,197 @@
+       return portPairGroup;
+   }
+
+   public PortPair createPortPair(PortPair portPair) {
+       checkArgument(portPair != null, "null passed for %s !", "Port Pair");
+       portPair = portPair.toBuilder().id(null).build();
+
+       portPair = this.osClient.sfc().portpairs().create(portPair);
+       return portPair;
+   }
+
+   public List<? extends FlowClassifier> listFlowClassifiers() {
+       return this.osClient.sfc().flowclassifiers().list();
+   }
+
+   public List<? extends PortPairGroup> listPortPairGroups() {
+       return this.osClient.sfc().portpairgroups().list();
+   }
+
+   public List<? extends PortPair> listPortPairs() {
+       return this.osClient.sfc().portpairs().list();
+   }
+
+   public List<? extends PortChain> listPortChains() {
+       return this.osClient.sfc().portchains().list();
+   }
+
+   public List<? extends Port> listPorts() {
+       return this.osClient.networking().port().list();
+   }
+
+   public List<? extends Port> listPorts(PortListOptions options) {
+       return this.osClient.networking().port().list(options);
+   }
+
+   public FlowClassifier getFlowClassifier(String flowClassifierId) {
+       return this.osClient.sfc().flowclassifiers().get(flowClassifierId);
+   }
+
+   public PortChain getPortChain(String portChainId) {
+       PortChain portChain = this.osClient.sfc().portchains().get(portChainId);
+       return fixPortChainCollections(portChain);
+   }
+
+   public PortPairGroup getPortPairGroup(String portPairGroupId) {
+       return this.osClient.sfc().portpairgroups().get(portPairGroupId);
+   }
+
+   public PortPair getPortPair(String portPairId) {
+       return this.osClient.sfc().portpairs().get(portPairId);
+   }
+
+   public Port getPort(String portId) {
+       return this.osClient.networking().port().get(portId);
+   }
+
+   public FlowClassifier updateFlowClassifier(String flowClassifierId, FlowClassifier flowClassifier) {
+       checkArgument(flowClassifierId != null, "null passed for %s !", "Flow Classifier Id");
+       checkArgument(flowClassifier != null, "null passed for %s !", "Flow Classifier");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       flowClassifier = flowClassifier.toBuilder().id(null).projectId(null).build();
+
+       return this.osClient.sfc().flowclassifiers().update(flowClassifierId, flowClassifier);
+   }
+
+   public PortChain updatePortChain(String portChainId, PortChain portChain) {
+       checkArgument(portChainId != null, "null passed for %s !", "Port Chain Id");
+       checkArgument(portChain != null, "null passed for %s !", "Port Chain");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portChain = portChain.toBuilder().id(null).projectId(null).chainParameters(null).chainId(null).build();
+
+       portChain = this.osClient.sfc().portchains().update(portChainId, portChain);
+       return fixPortChainCollections(portChain);
```

Comment:
we could potentially be overwriting the information from the port chain returned from openstack...is this needed?